src/submodule.c

```diff
@@ -1854,6 +1854,17 @@ static void submodule_release(git_submodule *sm)
 	git__free(sm);
 }
 
+int git_submodule_dup(git_submodule **out, git_submodule *source)
+{
+	GIT_ASSERT_ARG(out);
+
+	if (source != NULL)
```

Comment:
Yeah… `if (source)` would have been fine here. Sorry, another project has "all conditions must be explicit", so…

Reply:
There are a lot of occurrences of `!= NULL` in this file. Should I change it to implicit check?